src/Adapter/CMS/PageCategory/QueryHandler/GetCmsPageParentCategoryIdForRedirectionHandler.php

```diff
@@ -48,27 +47,11 @@ final class GetCmsPageParentCategoryIdForRedirectionHandler implements GetCmsPag
      */
     public function handle(GetCmsPageParentCategoryIdForRedirection $query)
     {
-        $parentId = CmsPageRootCategorySettings::ROOT_CMS_PAGE_CATEGORY_ID;
         try {
             $entity = new CMSCategory($query->getCmsPageCategoryId()->getValue());
-
-            if (0 >= $entity->id) {
-                throw new CmsPageCategoryNotFoundException(
-                    sprintf(
-                        'Unable to retrieve cms page category for redirection with id %s',
-                        $query->getCmsPageCategoryId()->getValue()
-                    )
-                );
-            }
-
             $parentId = (int) $entity->id_parent;
         } catch (PrestaShopException $e) {
-            throw new CmsPageCategoryException(
```

Comment:
Why do you remove the exception ? Is this an expected behavior ?

Reply:
yes there is a reason for it - query is on purpose called `GetCmsPageParentCategoryIdForRedirection` since on any error case it fallbacks to the default page. Ofcourse maybe I can do that on the controller, not sure which place is more appropriate. If I would to that on controller level then I think I should rename the query to `GetCmsPageParentCategoryId` since it will be applied for all use cases

Reply:
No it makes sense this way 👍